modules/packages/LinearAlgebra.chpl

```diff
@@ -1934,6 +1962,20 @@ module Sparse {
 
     return B;
   }
+  
+  proc _array.times(A: [?Adom] ?eltType) where isSparseArr(this) && !isCSArr(this)
+                                               && isSparseArr(A) && !isCSArr(A) {
+    if Adom.rank != this.domain.rank then compilerError("Unmatched ranks");
+    if this.domain.shape != Adom.shape then halt("Unmatched shapes");
+    var sps: sparse subdomain(Adom.parentDom);
+    sps += this.domain;
+    sps += Adom;
```

Comment:
I am fine with this as a good enough generic implementation. But you should add a comment saying that sps should actually be the intersection of two domains and not the union.

Reply:
I was pondering about several options on the behaviour of element-wise multiplication and division, on the set of indices in the returned matrix. 
We could either 
- Take the union: most easy to implement, no unexpected missing index, may waste space
- Follow CSR version (Add Adom to this if their index set don't match): I think users may not be aware of the logic and corresponding result unless they've looked into code 
- Take this.domain: users may expect index set to be based on `this`
- Take minimal index set: With some computation we can get the non-zero indices in the intersection of both domains, but `contains` operations may take (comparatively) a lot of time, and the behaviour would be very much different from the current CSR version

I'm inclining towards the first or last option

Reply:
Strictly for multiplication, I like the first option best, because of ease of implementation. I am not sure if there are many realistic use cases where you'd want to multiply to sparse arrays where their domains are not the same (I mean they have the same indices). Because otherwise, it feels to me that the resulting domain will be almost always empty.